File: fabric/src/main/java/cc/turtl/turtlshell/fabric/TurtlShellFabric.java
package cc.turtl.turtlshell.fabric;

import cc.turtl.turtlshell.TurtlShell;
import net.fabricmc.api.ClientModInitializer;

public final class TurtlShellFabric implements ClientModInitializer {
    @Override
    public void onInitializeClient() {
        EventRegisterFabric.register();
        TurtlShell.initClient();
    }
}

File: fabric/src/main/kotlin/cc/turtl/turtlshell/core/TurtlShellFabric.kt
package cc.turtl.turtlshell.core

import cc.turtl.turtlshell.TurtlShellCommon
import cc.turtl.turtlshell.example.core.ExampleCommon
import cc.turtl.turtlshell.impl.CommandGroupRegistrar
import net.fabricmc.api.ModInitializer
import net.fabricmc.fabric.api.command.v2.CommandRegistrationCallback

object TurtlShellFabric : ModInitializer {
    override fun onInitialize() {
        ExampleCommon.init()
        TurtlShellCommon.init()

        registerCommands()
    }

    private fun registerCommands() {
        CommandRegistrationCallback.EVENT.register { dispatcher, _, _ ->
            CommandGroupRegistrar.registerAll(dispatcher)
        }
    }
}